src/main/java/store/domain/promotion/Promotions.java

```diff
@@ -0,0 +1,26 @@
+package store.domain.promotion;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Promotions {
+    private final List<Promotion> promotions;
+
+    private Promotions(final List<Promotion> promotions) {
+        this.promotions = promotions;
+    }
+
+    public static Promotions from(final List<String> promotionData) {
+        List<Promotion> promotions = promotionData.stream()
+                .map(PromotionFactory::createPromotion)
+                .collect(Collectors.toList());
+        return new Promotions(promotions);
+    }
+
+    public Promotion findPromotion(final String productPromotion) {
+        return promotions.stream()
+                .filter(promotion -> promotion.isSamePromotion(productPromotion))
+                .findFirst()
+                .orElse(null);
```

Comment:
null보다는 NONE으로 명시해주면 NPE를 예방할 수 있을 것 같아요! 이 부분은 저도 코드리뷰를 받으며 알게 된 사실입니다😂
```suggestion
                .orElse(NONE);
```

Reply:
기존에는 프로모션이 적용되지 않을 때도 PromotionType enum에 적용 X인 값을 넣어놨었는데 리팩토링 하면서 그 부분을 빼고 아예 null을 가지게 했더니 생긴 부분이네요..! 확실히 null 보다는 그냥 기본 객체를 가지고 있게 하는 게 훨씬 좋겠어요